metadataStore/odm_templates.py

```diff
@@ -26,79 +27,111 @@ class BeginRunEvent(Document):
 
     Attributes
     ----------
+    uid: str
+        Globally unique id for this run
     time : timestamp
-        The date/time as found at the client side when an event is
+        The date/time as found at the client side when an begin_run_event is
         created.
-    beamline_id: str
+    beamline_id : str
         Beamline String identifier. Not unique, just an indicator of
         beamline code for multiple beamline systems
-    owner: str, optional
+    project : str, optional
+        Name of project that this run is part of
+    owner : str
         Specifies the unix user credentials of the user creating the entry
-    scan_id : int, optional
+    group : str
+        Unix group to associate this data with
+    scan_id : int
          scan identifier visible to the user and data analysis
-    beamline_config: bson.ObjectId, optional
+    beamline_config: bson.ObjectId
         Foreign key to beamline config corresponding to a given run
-    custom: dict, optional
-        Additional parameters that data acquisition code/user wants to
-        append to a given header. Name/value pairs
+    sample : dict
+        Information about the sample, may be a UID to another collection
     """
+    uid = StringField(required=True, unique=True)
```

Comment:
I agree with @tacaswell . We do need to add this on the ophyd side. @dchabot and @stuwilkins should decide what the format of this uid is. Thanks @ericdill  for the opyhd mds refactored api integration!!